### ISSUE TITLE
link_libraries() in parent cmake not working properly

### ISSUE THREAD

**Issue by farazrbx** [c-ares/c-ares#729]
I need to link an external library to be able to build c-ares on some platforms.
For some reason, using `link_libraries()` in my parent CMake file doesn't apply to c-ares, which is fetched with CMake's FetchContent.
However, using `list(APPEND CARES_DEPENDENT_LIBS mylib)` worked.
Is there anything in c-ares CMake that prevents `link_libraries()` from working?

**Comment by bradh352** [comment]
Can you show your actual usage?  Are you referencing  `cares::cares` as the dependency?

**Comment by farazrbx** [comment]
I have something like this in my project:

in `ThirdParty/c-ares/CMakeLists.txt`
```
set(CARES_STATIC ON)
set(CARES_SHARED OFF)
set(CARES_INSTALL OFF)
set(CARES_STATIC_PIC ON)

if(PLATFORM_X)
  link_libraries(some_extra_lib) # does not work
  list(APPEND CARES_DEPENDENT_LIBS some_extra_lib) # works
endif()

FetchContent_Declare(
  c-ares
  GIT_REPOSITORY      https://github.com/c-ares/c-ares
  GIT_TAG main
)

FetchContent_MakeAvailable(c-ares)
```

in my app's cmake:
```
add_library(myapp ....)
target_link_libraries(myapp PRIVATE c-ares::cares_static)
```

**Comment by bradh352** [comment]
I'm thinking this is something you might need to ask the CMake guys about.  I'm assuming FetchContent_MakeAvailable() isn't passing along the top level link_libraries()

**Comment by bradh352** [comment]
On a second look, its possible this line is causing the issue:
https://github.com/c-ares/c-ares/blob/main/CMakeLists.txt#L465

Likely we need to save the original value then restore it rather than just assuming it was blank.

**Comment by farazrbx** [comment]
Thanks!
Right. The top-level link_libraries() works fine with other third-party libs that I use.
Why do we need those 3 lines? We don't have them in curl's cmake.

**Comment by bradh352** [comment]
I'm not entirely sure, it looks like https://github.com/c-ares/c-ares/blob/main/CMakeLists.txt#L115 restores the original setting and I think that's the only thing that manipulates it.

**Comment by bradh352** [comment]
I think this is fixed in 4d8f538 ... if not, please re-open
